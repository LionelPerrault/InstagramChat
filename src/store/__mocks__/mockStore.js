import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'

// mock-reducers
import User from './reducers/User'
import Notification from './reducers/Notification'
import Group from './reducers/Group'
import Follow from './reducers/Follow'
import Explore from './reducers/Explore'
import Post from './reducers/Post'
import Hashtag from './reducers/Hashtag'
import Setting from './reducers/Setting'
import Message from './reducers/Message'

const initialState = {
  User,
  Notification,
  Group,
  Follow,
  Explore,
  Post,
  Hashtag,
  Setting,
  Message,
}

const middlewares = [thunk]
const createStore = configureStore(middlewares)
const mockStore = createStore(initialState)

export default mockStore
