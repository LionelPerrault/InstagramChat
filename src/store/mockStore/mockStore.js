import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import logger from 'redux-logger'
import promise from 'redux-promise-middleware'

// mock-reducers
import User from './mock-reducers/User'
import Notification from './mock-reducers/Notification'
import Group from './mock-reducers/Group'
import Follow from './mock-reducers/Follow'
import Explore from './mock-reducers/Explore'

const initialState = {
  User,
  Notification,
  Group,
  Follow,
  Explore
}

const middlewares = [
  promise(), thunk, logger
]
const createStore = configureStore(middlewares)
const mockStore = createStore(initialState)

export default mockStore
