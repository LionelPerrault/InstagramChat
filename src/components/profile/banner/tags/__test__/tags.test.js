import React from 'react'
import { create } from 'react-test-renderer'
import { Provider } from 'react-redux'
import BannerTags from '../tags'
import MockDataElement from '../../../../../utils/__mocks__/mock-dataElement'
import mockStore from '../../../../../store/__mocks__/mockStore'
import { shallow } from 'enzyme'

describe('BannerTags Component', () => {
  MockDataElement()

  it('should match snapshot', () => {
    const tree = create(
      <Provider store={mockStore}>
        <BannerTags />
      </Provider>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should show tags when toggle icon is clicked', () => {
    const wrapper = shallow(<BannerTags />)
    wrapper.find('.pro_exp_more').simulate('click')
    expect(wrapper.find('.pro_tags').exists()).toBe(true)
  })
})
