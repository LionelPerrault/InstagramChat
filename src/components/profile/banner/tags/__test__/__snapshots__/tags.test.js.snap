// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BannerTags Component should match snapshot 1`] = `
<div
  className="pro_exp_more"
  onClick={[Function]}
>
  <span
    data-tip="Tags"
  >
    <i
      className="material-icons"
    >
      expand_more
    </i>
  </span>
</div>
`;
